lib/plugins/features/movies/controller.js

```diff
@@ -3,6 +3,11 @@
 const Movie = require('../../../models/movie');
 
 exports.create = (payload) => {
+  if (payload.hasOwnProperty('title')) {
```

Comment:
Oooh, interesting. I don't think I've ever seen anyone take a non-breaking change to this interface.

A couple nits:

For this case, you can just do:
```js
if (payload.title) {
  payload.name = payload.title;
  delete payload.title;
}
```

That said, careful about deleting fields. Because javascript passes references to objects, deleting can cause some really weird bugs. For example, when a controller deletes a field of the payload object, that data is lost to logging. 

I recommend cloning the input object if you're planning on manipulating the fields. 
```javascript
payload = Object.assign({}, payload);

if (payload.title) {
  payload.name = payload.title;
  delete payload.title;
}
```

Reply:
Another thing you're going to want to do is update the validator to not allow both `title` and `name` to be passed in. This can be done with JOI's [`nand`](https://github.com/hapijs/joi/blob/v11.0.1/API.md#objectnandpeers).

Reply:
Though also, you're more than welcome to just not support the breaking change and change the validator to only accept `name`.

Reply:
I had delete initially, but the linter complains about it:

> 10:5  error  Avoid using the delete keyword, instead use Reflect.deleteProperty  prefer-reflect

Since name is not a valid key defined in my validator, do I need to still check with a nand? Maybe I'm misunderstanding, but I thought the public API is still 'title' but internally at the controller and model layers it's 'name'. So at the time it goes through the validator, name should never be defined and if it is, it'll error since it's not a valid key in my validator?

Reply:
Per discussions IRL the prompt for [step 2](https://www.notion.so/lob/Zero-Downtime-Schema-Changes-c9b505f6c87145b384124218c9243860#c857cb4024714285a98ef1b64bec6de2) says the public API interface should remain the same:

> In this case, let's assume we want to change the column name in the database, but keep the same interface for the API. The API should still accept the title parameter.

So I don't think the validating 'name' makes sense as it should not be part of the API interface.